src/main/java/com/unosquare/cvgenerator/controller/HelloController.java

```diff
@@ -1,29 +1,53 @@
 package com.unosquare.cvgenerator.controller;
 
 import com.unosquare.cvgenerator.model.dto.GreetingDTO;
+import com.unosquare.cvgenerator.model.dto.TemplateDetailDTO;
 import com.unosquare.cvgenerator.model.view.GreetingView;
 import com.unosquare.cvgenerator.service.GreetingService;
+import com.unosquare.cvgenerator.service.TemplateService;
 import com.unosquare.cvgenerator.util.MapperUtil;
 import lombok.RequiredArgsConstructor;
-import org.springframework.stereotype.Controller;
-import org.springframework.ui.Model;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
 
-@Controller
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Slf4j
+@RestController
 @RequiredArgsConstructor
 public class HelloController {
 
     private final GreetingService greetingService;
+    private final TemplateService templateService;
     private final MapperUtil mapperUtil;
 
     @GetMapping("/hello/{id}")
-    public String sayHello(@PathVariable("id") Integer id, Model theModel) {
+    public String sayHello(@PathVariable("id") Integer id) {
+
+        // Get greeting and template
         GreetingDTO greetingDTO = greetingService.findById(id);
         GreetingView greetingView = mapperUtil.map(greetingDTO, GreetingView.class);
-        theModel.addAttribute("greetingView", greetingView);
+        TemplateDetailDTO templateDetailDTO = templateService.findById(id);
+        String template = templateDetailDTO.getTemplateHtml();
+
+        // Build regex replacement map
+        HashMap<String, String> keyTermsMap = new HashMap<>();
+        keyTermsMap.put("\\B%%greeting%%\\B", greetingView.getGreeting());
+
+        // Perform the string replacement
+        String result = "";
+        for (Map.Entry<String, String> entry : keyTermsMap.entrySet()) {
```

Comment:
I'd extract this out into a service for replacing the values.

You could also make this much more efficient, but we can work on that later.